src/routes/write-exercise/components/InteractionSection.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { Keyboard, Pressable, TouchableOpacity } from 'react-native'
+import { heightPercentageToDP as hp } from 'react-native-responsive-screen'
+import stringSimilarity from 'string-similarity'
+import styled from 'styled-components/native'
+
+import { CloseIcon } from '../../../../assets/images'
+import AudioPlayer from '../../../components/AudioPlayer'
+import Button from '../../../components/Button'
+import { BUTTONS_THEME, numberOfMaxRetries, SIMPLE_RESULTS, SimpleResultType } from '../../../constants/data'
+import labels from '../../../constants/labels.json'
+import { COLORS } from '../../../constants/theme/colors'
+import { DocumentResultType } from '../../../navigation/NavigationTypes'
+import { stringifyDocument } from '../../../services/helpers'
+import Feedback from './Feedback'
+import MissingArticlePopover from './MissingArticlePopover'
+
+const TextInputContainer = styled.View<{ styledBorderColor: string }>`
+  width: 80%;
+  height: ${hp('8%')}px;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  border-radius: 2px;
+  padding: 0 15px;
+  margin-bottom: 5%;
+  border: 1px solid ${prop => prop.styledBorderColor};
+`
+const StyledTextInput = styled.TextInput`
+  flex: 1;
+  font-size: ${props => props.theme.fonts.largeFontSize};
+  font-weight: ${props => props.theme.fonts.lightFontWeight};
+  letter-spacing: ${props => props.theme.fonts.listTitleLetterSpacing};
+  font-family: ${props => props.theme.fonts.contentFontRegular};
+  color: ${prop => prop.theme.colors.lunesBlack};
+  width: 90%;
+`
+
+export const LightLabelInput = styled.Text<{ styledInput?: string }>`
+  text-align: center;
+  font-family: ${props => props.theme.fonts.contentFontBold};
+  font-size: ${props => props.theme.fonts.defaultFontSize};
+  letter-spacing: ${props => props.theme.fonts.capsLetterSpacing};
+  text-transform: uppercase;
+  font-weight: ${props => props.theme.fonts.defaultFontWeight};
+  color: ${prop =>
+    prop.styledInput ? props => props.theme.colors.lunesBlackLight : props => props.theme.colors.lunesWhite};
+`
+
+const Speaker = styled.View`
+  top: -20px;
+`
+
+interface InteractionSectionProps {
+  documentWithResult: DocumentResultType
+  isAnswerSubmitted: boolean
+  storeResult: (result: DocumentResultType) => void
+}
+
+const almostCorrectThreshold = 0.6
+const ttsThreshold = 0.6
+
+const InteractionSection = (props: InteractionSectionProps): ReactElement => {
+  const { isAnswerSubmitted, documentWithResult, storeResult } = props
+
+  const [isArticleMissing, setIsArticleMissing] = useState<boolean>(false)
+  const [input, setInput] = useState<string>('')
+  const [submittedInput, setSubmittedInput] = useState<string>('')
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+
+  const editable = !isAnswerSubmitted || documentWithResult.result === 'similar'
```

Comment:
Rename
```suggestion
  const retryAllowed = !isAnswerSubmitted || documentWithResult.result === 'similar'
```
and use it below (line 179)